yahttp/reqresp.hpp

```diff
@@ -168,9 +148,7 @@ namespace YaHTTP {
     ssize_t max_request_size; //<! maximum size of request
     ssize_t max_response_size;  //<! maximum size of response
     bool is_multipart; //<! if the request is multipart, prevents Content-Length header
-#ifdef HAVE_CPP_FUNC_PTR
-    funcptr::function<size_t(const HTTPBase*,std::ostream&,bool)> renderer; //<! rendering function
-#endif
+    std::function<size_t(const HTTPBase*,std::ostream&,bool)> renderer; //<! rendering function
```

Comment:
this commit should be split into two:

 - 1st - change the optional => mandatory
 - 2nd - do the std::function changes